### PR TITLE
feat(Search): allow tracking Enter key press

### PR DESCRIPTION
**Pull request by YossiSaadi**
https://monday.monday.com/boards/3532714909/pulses/7890250116

### CODE REVIEW

File: packages/core/src/components/Search/Search.tsx [hunk 1]
```diff
@@ -35,6 +35,7 @@ const Search = forwardRef(
       onFocus,
       onBlur,
       onClear,
+      onEnterKey,
```
Comment (shlomitc) [inline packages/core/src/components/Search/Search.tsx:38]: why aren't we just exposing `onKeyDown`?

Comment (YossiSaadi) [reply packages/core/src/components/Search/Search.tsx:38]: I actually thought about doing it, but maybe we should make this more straightforward for cases like Enter, which is the one the team currently needs. I can't think of another key that someone might want to check for stuff like Search field

File: packages/core/src/components/Search/Search.tsx [hunk 2]
```diff
@@ -53,14 +54,11 @@ const Search = forwardRef(
...
-      if (disabled) {
-        return;
-      }
-
+      if (disabled) return;
       inputRef.current?.focus?.();
       clearValue();
       onClear?.();
-    }, [disabled, clearValue]);
+    }, [disabled, clearValue, onClear]);
```
Comment (talkor) [inline packages/core/src/components/Search/Search.tsx:61]: Needed?

**Review by talkor — APPROVED** [review]

File: packages/core/src/components/Search/Search.tsx (hunk 1 above)
Comment (YossiSaadi) [reply packages/core/src/components/Search/Search.tsx]: @shlomitc after reconsideration, went with your suggestion at the end, thanks!

**Pull request merged**